### DISCUSSION TITLE
Release 4.0 with Xamarin Support? (iOS, Android and UWP)

### DISCUSSION THREAD

**Discussion by Agredo** [Mapsui/Mapsui#1597] [General]
I checked the Release notes. 
Will the new version 4.0 drop xamarin.forms support? 
It would be sad if it now drop UWP support because WinUI3 is currently not a full replacement for UWP. And when there will be Xamarin:Forms support (.Net Standard 2.0) then why not UWP? 

Thank you :)

**Comment by pauldendulk** [comment]
I think you are asking if Mapsui 4.0 support Xamarin.Forms UWP. Is this correct?

If so, the answer is that it will be supported. Xamarin.Forms is supported and UWP is a part of that.

We also shipped a nuget specifically for UWP and that is what we are dropping. See here, there are no new beta releases:
https://www.nuget.org/packages/Mapsui.uwp

Notes:
- *Maintenance costs are high*: The costs for supporting a platform are high, and underestimated by everyone I talk to. The costs are not in the code itself, but are in the maintenance around it (testing, getting it to build on the build server, keep compatibility with other parts, answering questions, issues related to some specific device).
- *More platforms dropped in v5*: I think Mapsui should move fully to .NET 6 and SDK style csprojs. Microsoft itself is not working with older projects and the tooling for those older platforms is not developed anymore and we need to take advantage of the progress to keep up.
- *Keep the timelines in mind*: It will take some more months before v4 is released, and it will take some time before users will start updating to v4, and it will take some time for users to release their apps with v4. By the time you get there the world has changed.
- *You don't have to go to v4*: For many users there is no reason to go to v4. They can hang on to v3 for a while, before making the jump to for instance Maui.

**Reply by Agredo** [reply]
Yes exactly! I ment Xamairn.Forms Support with UWP. I thought you are having UWP custom renderers. But when it works perfekt! I can understand and its okay to drop UWP for WinUI but currently WinUI3 is not a real replacement for UWP. 

Thank you for your answer!

**Reply by pauldendulk** [reply]
ah okay. I updated the release notes to be more clear about this https://github.com/Mapsui/Mapsui/releases/tag/4.0.0-beta.1
